src/Microsoft.DotNet.Wpf/src/Shared/cpp/dwriteloader.cpp

```diff
@@ -7,37 +7,9 @@
 namespace WPFUtils
 {
 
-#if defined(__cplusplus_cli)
-#endif
 HMODULE LoadDWriteLibraryAndGetProcAddress(void **pfncptrDWriteCreateFactory)
 {
-    HMODULE hDWriteLibrary = nullptr;
-    
-    // KB2533623 introduced the LOAD_LIBRARY_SEARCH_SYSTEM32 flag. It also introduced
-    // the AddDllDirectory function. We test for presence of AddDllDirectory as an 
-    // indirect evidence for the support of LOAD_LIBRARY_SEARCH_SYSTEM32 flag. 
-    HMODULE hKernel32 = GetModuleHandle(L"kernel32.dll");
-    if (hKernel32 != nullptr)
-    {
-        if (GetProcAddress(hKernel32, "AddDllDirectory") != nullptr)
-        {
-            // All supported platforms newer than Vista SP2 shipped with dwrite.dll.
-            // On Vista SP2, the .NET servicing process will ensure that a MSU containing 
-            // dwrite.dll will be delivered as a prerequisite - effectively guaranteeing that 
-            // this following call to LoadLibraryEx(dwrite.dll) will succeed, and that it will 
-            // not be susceptible to typical DLL planting vulnerability vectors.
-            hDWriteLibrary = LoadLibraryEx(L"dwrite.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
-        }
-        else 
-        {
-            // LOAD_LIBRARY_SEARCH_SYSTEM32 is not supported on this OS. 
-            // Fall back to using plain ol' LoadLibrary
-            // There is risk that this call might fail, or that it might be
-            // susceptible to DLL hijacking. 
-            hDWriteLibrary = LoadLibrary(L"dwrite.dll");
-        }
-    }
-    
+    HMODULE hDWriteLibrary = LoadLibraryEx(L"dwrite.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
```

Comment:
Windows 7, Windows Server 2008 R2, Windows Vista and Windows Server 2008:  This value requires [KB2533623](https://support.microsoft.com/kb/2533623) to be installed. See https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa

Reply:
Argh! I miss that. But only Windows 7 SP1 is possible applicable since other OS [does not supported](https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md) by .NET 7
And even that seems to be [KB2533623](https://support.microsoft.com/kb/2533623) required for .NET see https://github.com/dotnet/docs/issues/20459

But I hardly can understand is ESU has this patch or not.

Reply:
That should be slated for .NET 8 then. That way Win 7 would be out of support at release date (Nov 2023) and not important anymore.